tests/h/search/query_test.py

```diff
@@ -141,6 +141,74 @@ def search(self, search):
 
 
 class TestUriFilter(object):
+    def test_filters_by_uri(self, search, Annotation):
+        Annotation(target_uri="http://bar.com")
+        Annotation(target_uri="https://bar.com")
+        Annotation(target_uri="https://foo/bar.com")
+        expected_ids = [Annotation(target_uri="bar.com").id]
+
+        result = search.run({"uri": "bar.com"})
+
+        assert sorted(result.annotation_ids) == sorted(expected_ids)
+
+    def test_filters_by_url(self, search, Annotation):
+        Annotation(target_uri="http://bar.com")
+        Annotation(target_uri="https://bar.com")
+        Annotation(target_uri="https://foo/bar.com")
+        expected_ids = [Annotation(target_uri="bar.com").id]
+
+        result = search.run({"url": "bar.com"})
+
+        assert sorted(result.annotation_ids) == sorted(expected_ids)
+
+    def test_filters_on_whole_url(self, search, Annotation):
+        Annotation(target_uri="http://bar.com")
```

Comment:
I think a more representative test here would be that given annotations on say `http://site.com/` and `http://site.com/article`, a query for `http://site.com` will only return the first annotation. In other words, the homepage will only show annotations for the homepage and not subpages.